tools/src/commands/PublishProjectTemplates.ts

```diff
@@ -126,31 +141,39 @@ async function action(options) {
 
     const moreArgs: string[] = [];
 
-    if (tag) {
-      // Assign custom tag in the publish command, so we don't accidentally publish as latest.
-      moreArgs.push('--tag', tag);
-    }
+    // Assign custom tag in the publish command, so we don't accidentally publish as latest.
+    moreArgs.push('--tag', npmPublishTag);
 
     // Publish to NPM registry
+    console.log('Executing command: npm publish ' + ['--access', 'public', ...moreArgs].join(' '));
     options.dry ||
       (await spawnAsync('npm', ['publish', '--access', 'public', ...moreArgs], {
         stdio: 'inherit',
         cwd: template.path,
       }));
 
-    if (tag && (await shouldAssignLatestTagAsync(template.name, newVersion))) {
+    if (tags.length > 1) {
+      // Additional tag (latest, beta, or next) is added here
       console.log(
-        `Assigning ${chalk.blue('latest')} tag to ${chalk.green(template.name)}@${chalk.red(
+        `Assigning ${chalk.blue(`${tags[1]}`)} tag to ${chalk.green(template.name)}@${chalk.red(
           newVersion
         )}...`
       );
 
-      // Add the latest tag to the new version
+      // Add the tag to the new version
+      console.log(
+        'Executing command: npm ' +
```

Comment:
this scared me when i did a dry run 😂 maybe we should make it clear that it's not actually executing and it is respecting the dry flag

Reply:
Fixed